### PR TITLE
[SMTChecker] Fix internal error in tuples of tuples.

### LINKED ISSUE #8971: [SMTChecker] ICE in solidity::frontend::SMTEncoder::assignment(const solidity::frontend::Expression &, const vector<smt::Expression> &, const solidity::frontend::TypePointer &, const langutil::SourceLocation &)
## Description

This contract:
```solidity
pragma experimental SMTChecker;
contract C {
    function f3() public returns(int) {
        ((, ), ) = ((7, 8), 9);
    }
}
```

produces:

```
Internal compiler error during compilation:
/home/user/solidity/libsolidity/formal/SMTEncoder.cpp(1291): Throw in function void solidity::frontend::SMTEncoder::assignment(const solidity::frontend::Expression &, const vector<smt::Expression> &, const solidity::frontend::TypePointer &, const langutil::SourceLocation &)
Dynamic exception type: boost::exception_detail::clone_impl<solidity::langutil::InternalCompilerError>
std::exception::what: 
[solidity::util::tag_comment*] = 
```

when compiled with solc

On master, using AFL fuzzing.  Another discovery using https://github.com/agroce/afl-compiler-fuzzer.  I threw ALL the .sol files under libsolidity into the corpus, so it's effectively fuzzing the SMTChecker and such now, also, I expect this will turn up some things.

## Environment

- Compiler version:  0.6.8-develop.2020.5.13+commit.1700bdc4.Linux.clang
- Target EVM version (as per compiler settings): N/A
- Framework/IDE (e.g. Truffle or Remix): N/A
- EVM execution environment / backend / blockchain client: N/A
- Operating system: Ubuntu 18.04 in docker

## Steps to Reproduce

Above shows pretty clearly, I think.

### PR DESCRIPTION
**Pull request by leonardoalt**
Fixes https://github.com/ethereum/solidity/issues/8971

### CODE REVIEW

**Comment by leonardoalt** [comment]
Added test

File: Changelog.md [hunk 1]
```diff
@@ -28,6 +28,8 @@ Bugfixes:
  * SMTChecker: Fix internal error when applying arithmetic operators to fixed point variables.
  * SMTChecker: Fix internal error when short circuiting Boolean expressions with function calls in state variable initialization.
  * SMTChecker: Fix internal error when assigning to index access inside branches.
+ * SMTChecker: Fix internal error when encoding tuples of tuples.
```
Comment (ekpyron) [inline Changelog.md]: This will need to move.

**Review by ekpyron — COMMENTED** [review]
Fine with merging this - will approve once the release is through and the changelog sorted out!

**Comment by leonardoalt** [comment]
@ekpyron rebased & fixed

File: Changelog.md [hunk 2]
```diff
@@ -37,6 +45,12 @@ Bugfixes:
  * Type Checker: Fix internal compiler error when forward referencing non-literal constants from inline assembly.
  * Type Checker: Fix internal compiler error when trying to decode too large static arrays.
  * Type Checker: Fix wrong compiler error when referencing an overridden function without calling it.
+ * NatSpec: DocString block is terminated when encountering an empty line.
+ * Scanner: Fix bug when two empty NatSpec comments lead to scanning past EOL.
+ * Code Generator: Trigger proper unimplemented errors on certain array copy operations.
+ * SMTChecker: Fix internal error when applying arithmetic operators to fixed point variables.
+ * SMTChecker: Fix internal error when short circuiting Boolean expressions with function calls in state variable initialization.
+ * SMTChecker: Fix internal error when assigning to index access inside branches.
```
Comment (ekpyron) [inline Changelog.md]: This doesn't look right...

**Review by ekpyron — CHANGES_REQUESTED** [review]

File: Changelog.md (hunk 2 above)
Comment (leonardoalt) [reply Changelog.md]: Damn, wait

Comment (leonardoalt) [reply Changelog.md]: done

**Review by ekpyron — APPROVED** [review]

**Pull request merged**